sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -3173,6 +3174,41 @@ class CollationSQLExpressionsSuite
     })
   }
 
+  test("from_json and from_xml transformations with session collation") {
+    val spark = this.spark
+    import spark.implicits._
+
+    def checkSchema(
+        dataset: Dataset[String],
+        transformation: Column,
+        expectedSchema: StructType): Unit = {
+      val transformedSchema = dataset.select(transformation.as("result")).schema
+      assert(transformedSchema === expectedSchema)
+    }
+
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> "UNICODE_CI_AI") {
+      Seq(
+        StringType,
+        StringType("UTF8_BINARY"),
+        StringType("UNICODE"),
+        StringType("UNICODE_CI_AI"),
+      ).foreach { stringType =>
+        val dataSchema = StructType(Seq(StructField("fieldName", stringType)))
+        val expectedSchema = StructType(Seq(StructField("result", dataSchema)))
+
+        // JSON Test
+        val jsonData = Seq("""{"fieldName": "fieldValue"}""")
+        val jsonDataset = spark.createDataset(jsonData)
+        checkSchema(jsonDataset, from_json($"value", dataSchema), expectedSchema)
```

Comment:
Just in case, the common approach in Spark SQL is to place function tests to `*FunctionsSuite` and expression tests to `ExpressionsSuite`, see:
- JsonFunctionsSuite
- JsonExpressionsSuite

How about to create new test suite `CollationSQLFunctionsSuite`, and place the new test into it.

Reply:
makes sense, done!